core/unit_test/TestUniqueToken.hpp

```diff
@@ -129,6 +127,116 @@ class TestUniqueToken {
   }
 };
 
-TEST(TEST_CATEGORY, unique_token) { TestUniqueToken<TEST_EXECSPACE>::run(); }
+TEST(TEST_CATEGORY, unique_token_global) {
+  TestUniqueToken<TEST_EXECSPACE,
+                  Kokkos::Experimental::UniqueTokenScope::Global>::run();
+}
+
+TEST(TEST_CATEGORY, unique_token_instance) {
+  TestUniqueToken<TEST_EXECSPACE,
+                  Kokkos::Experimental::UniqueTokenScope::Instance>::run();
+}
+
+template <class Space>
+class TestAcquireTeamUniqueToken {
+ public:
+  using execution_space = typename Space::execution_space;
+  using view_type       = Kokkos::View<int*, execution_space>;
+  using scratch_view =
+      Kokkos::View<int, typename execution_space::scratch_memory_space,
+                   Kokkos::MemoryUnmanaged>;
+  using team_policy_type = Kokkos::TeamPolicy<execution_space>;
+  using team_member_type = typename team_policy_type::member_type;
+  using tokens_type      = Kokkos::Experimental::UniqueToken<execution_space>;
+
+  tokens_type tokens;
+
+  view_type verify;
+  view_type counts;
+  view_type errors;
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(team_member_type team) const {
+    Kokkos::Experimental::AcquireTeamUniqueToken<team_policy_type> token_val(
+        tokens, team);
+    scratch_view team_rank_0_token_val(team.team_scratch(0));
+    const int32_t t = token_val.value();
+
+    bool ok = true;
+
+    ok = ok && 0 <= t;
+    ok = ok && t < tokens.size();
```

Comment:
Do these need to be atomic_and?

Reply:
No: this error check happens in each thread independently.